### PR TITLE
Feat/181 delete member modal

### LINKED ISSUE #181: [FEAT] Organization settings  Manage Members > Implement this Delete Member modal
## Description
Implement the "Delete Member" modal on the member page.

## Acceptance Criteria
- The "Delete Member" modal should have:
  - A heading "Delete Member" properly aligned
  - A message: "Are you sure you want to delete Chad Bosewick? All of your data will be permanently removed. This action cannot be undone."
  - Two buttons at the bottom: "Cancel" and "Delete"
- Clicking outside the modal and clicking the times button should close the modal
- Proper usage of overlay opacity on the parent page when the modal is open
- The modal adheres to the brand colors and design
- The modal is responsive across device widths

## Expected Outcome
- User should see a properly aligned "Delete Member" modal with the specified text and buttons
- Clicking "Cancel" should close the modal
- Clicking "Delete" should delete the member and close the modal
- Clicking outside the modal or the times button should close the modal
- The modal should be responsive

## Purpose
As an administrator, I want to delete members from the workspace with a clear confirmation process.

## Requirement
- Implement the "Delete Member" modal with the specified text and buttons
- Ensure the modal closes when clicking outside it or the times button
- Ensure overlay opacity is applied to the parent page when the modal is open
- Ensure the modal follows the brand's design guidelines
- Make sure the modal is responsive

## Testing
- Test if the "Delete Member" modal displays correctly
- Test if the heading, message, and buttons are properly aligned
- Test if clicking "Cancel" closes the modal
- Test if clicking "Delete" deletes the member and closes the modal
- Test if clicking outside the modal or the times button closes the modal
- Test if overlay opacity is applied correctly when the modal is open
- Test responsiveness of the modal on various device widths

## Task
- [ ] Create "Delete Member" modal with specified heading, message, and buttons
- [ ] Implement functionality to close the modal when clicking outside it or the times button
- [ ] Ensure overlay opacity is applied to the parent page when the modal is open
- [ ] Ensure the modal follows brand colors and design
- [ ] Ensure the modal is responsive

## Parent Page Component Issue Link
[Link](https://github.com/hngprojects/hng_boilerplate_nextjs/issues/52)

## Picture
Confirm Delete

## Dev Mode
- [Dev Mode Link](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=96-100990&m=dev)

## Timeframe
 - Friday, 8am

### PR DESCRIPTION
**Pull request by onyinyechi-okeke**
# Description
closes #181 

# Changes proposed

## What were you told to do?
I was required to create [FEAT] Organizations Members > implement delete member modal

## What did you do?
I created the deletemodal under components/modals 
Made it responsive
Created a custom hook under hooks to be accesible to my parent page
Wrote tests for the component under tests

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

Screenshot (201)

### CODE REVIEW

File: src/hooks/useDeleteModal.tsx
Comment (JulianaSau) [inline src/hooks/useDeleteModal.tsx:1]: Why do you need this hook? is this component being used in more than one place?

**Comment by onyinyechi-okeke** [comment]
Yes, it will also be used in the parent component. The parent component
will add an onclick function so as to be able to open my modal and then
pass on the closemodal function to my component.

**Comment by onyinyechi-okeke** [comment]
It will only be used in the parent component but I don’t know how else the
parent component will access the hook so as to be able to open the modal

File: src/hooks/useDeleteModal.tsx
Comment (onyinyechi-okeke) [reply src/hooks/useDeleteModal.tsx:1]: I just deleted the hook and defined my state in the component instead

**Comment by onyinyechi-okeke** [comment]
Screenshot (206)
Screenshot (205)

**Comment by oreoluwa212** [comment]
Test failed, update with dev and test

**Pull request closed without merging**